modules/contact/doc/content/source/constraints/MechanicalContactConstraint.md

```diff
@@ -1 +1,57 @@
+
+### Contact Formulations
+
+To enforce the contact constraints, three types of formulations are available. These are `kinematic`, `penalty`, and `mortar`.
+The `kinematic` and `penalty` formulations have been developed for some time and turn out to be robust for many mechanical contact problems. Recent efforts have been put in developing a mortar-based contact formulation. In this documentation, however,  we focus on `kinematic` and `penalty` formulations. For readers who are interested in mortar-based mechanical contact, please go to [mortar constraint system](Constraints/index.md) for more information.
+
+The overall workflow of imposing a mechanical constraint is as follows: 1) Determine whether penetration occurs. If so, constrain displacement; 2) In the normal direction, eliminate the penetration; 3) In the tangential direction, slip must satisfy friction model, no slit is allowed for the `glued` model, no constraint is applied for the `frictionless` model, Kuhn-Tucker conditions must be satisfied for the `coulomb` model. The mechanical contact enforcement steps for the `glued` and `frictionless` models are summarized below:
+
+For `penalty`:
+
+|  Glued  | Frictionless |
+| ---------- | ------------ |
+|  $\boldsymbol{f}_c = k_p (\boldsymbol{x} - \boldsymbol{x}_{p_{t-1}})$     | $\boldsymbol{f}_c = \boldsymbol{n}\boldsymbol{n}^{T} k_p (\boldsymbol{x} - \boldsymbol{x}_p)$ |
+|  $\boldsymbol{r}_s = \boldsymbol{r}_s + \boldsymbol{f}_c$    | $\boldsymbol{r}_s = \boldsymbol{r}_s + \boldsymbol{f}_c$ |
+|  $\boldsymbol{r}_m = \boldsymbol{r}_m - \phi_i\boldsymbol{f}_c$     | $\boldsymbol{r}_m = \boldsymbol{r}_m - \phi_i \boldsymbol{f}_c$ |
+
+For `kinematic`:
+
+|  Glued  | Frictionless |
+| ---------- | ------------ |
+|  $\boldsymbol{f}_c = -\boldsymbol{r}_{s_{copy}}$     | $\boldsymbol{f}_c = -\boldsymbol{n}\boldsymbol{n}^{T} \boldsymbol{r}_{s_{copy}}$  |
+|  $\boldsymbol{r}_s = \boldsymbol{r}_s+\boldsymbol{f}_c+k_p(\boldsymbol{x} - \boldsymbol{x}_{p_{t-1}})$     | $\boldsymbol{r}_s = \boldsymbol{r}_s + \boldsymbol{f}_c + \boldsymbol{n}\boldsymbol{n}^{T} k_p (\boldsymbol{x} - \boldsymbol{x}_{p_{t-1}})$ |
+|  $\boldsymbol{r}_m = \boldsymbol{r}_m - \phi_i \boldsymbol{f}_c$     | $\boldsymbol{r}_m = \boldsymbol{r}_m - \phi_i \boldsymbol{f}_c$ |
+
+Note the kinematic method uses penalty parameter only to penalize error. Therefore, the converged solution has no error due to penalty compliance
+
+For the frictional contact, the `penalty` method calculates normal force and tangential predictor force using penalty stiffness and check for frictional capacity. While using the `kinematic` method, the deformation of surrounding material serves the role of the penalty parameter. The problem is solved using nested solves. The normal and tangential contact tractions are computed based on strict enforcement of glued constraint. The diagonalized stiffness of slave nodes are used as regularization parameter to calculate slip.
```

Comment:
Drop everything starting with 'The problem is solved using nested solves.' That was previously the case, but not any more.

Reply:
Thanks @bwspenc ! Nice catch!